templates/ovncontroller/bin/liveness.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/sh
```

Comment:
Finally, we removed set -e to capture a specific error message when $? is not equal to 0. This way, the output includes not only our custom error message, but also the output from the failed command. Let me know if you consider any further option.

Reply:
This is because you should not check `$?` but use `! cmd` syntax. See:

```
$ cat /tmp/test.sh
#!/bin/bash

set -e

error_exit() {
	echo "Logging!"
	exit 1
}

if ! false; then
	error_exit
fi

$ cat /tmp/test2.sh
#!/bin/bash

set -e

error_exit() {
	echo "Logging!"
	exit 1
}

false
if [ $? -ne 0 ]; then
	error_exit
fi

$ bash /tmp/test.sh
Logging!

$ bash /tmp/test2.sh
$
```

Reply:
This is documented in the official bash docs. See: https://www.gnu.org/software/bash/manual/html_node/The-Set-Builtin.html

```
The shell does not exit if the command that fails is part of the command list immediately following a while or until keyword, part of the test in an if statement, part of any command executed in a && or || list except the command following the final && or ||, any command in a pipeline but the last, or if the command’s return status is being inverted with !.
```

Reply:
I've noticed your script (claims to be) POSIX shell (`/bin/sh`), so I figured I should check [POSIX standard](https://pubs.opengroup.org/onlinepubs/007904875/utilities/set.html).

```
-e
    When this option is on, if a simple command fails for any of the reasons listed in [Consequences of Shell Errors](https://pubs.opengroup.org/onlinepubs/007904875/utilities/xcu_chap02.html#tag_02_08_01) or returns an exit status value >0, and is not part of the compound list following a while, until, or if keyword, and is not a part of an AND or OR list, and is not a pipeline preceded by the ! reserved word, then the shell shall immediately exit.
```

Reply:
Switching to bash since the simplicity of the script allows it, and it seems that certain flags can cause issues (e.g. pidof -q O_o) after reading some posts. Didn't know the `! cmd` and looks very clean, so I don’t have to keep checking `$?`